File: py/ros2_pubsub_demo/pub.py
from time import sleep
import rclpy
from std_msgs.msg import String


if __name__ == "__main__":
    print("ros py pub")
    rclpy.init()

    node = rclpy.create_node("py_pub")

    publisher = node.create_publisher(String, 'topic', 10)

    msg = String()

    i = 0
    while rclpy.ok():
        msg.data = 'Hello World: %d' % i
        print("pub:",msg)
        i += 1
        publisher.publish(msg)
        sleep(0.5)  # seconds

    node.destroy_node()
    rclpy.shutdown()

File: cpp/ros/ros_pubsub_demo/pub.cc
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char** argv) {
  std::cout << "Hello, Ros Pub!" << std::endl;
  ros::init(argc, argv, "pub");
  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<std_msgs::String>("/chat", 1);
  ros::Rate r(5);
  std_msgs::String msg;
  msg.data = "hello";
  while (ros::ok()) {
    pub.publish(msg);
    r.sleep();
  }
  return 0;
}
